examples/spacecraft.jl

```diff
@@ -0,0 +1,382 @@
+
+    #mode 2 ("rendezvous attempt")
+    invariant = HalfSpace(sparsevec([x], [-1.], n), 100.) # x >= -100
+    attempt = @system(x' = spacecraft_attempt!(x), dim:5, x ∈ invariant)
+
+    #mode 3 "aborting"
+    invariant = Universe(n)
+    aborting = @system(x' = spacecraft_aborting!(x), dim: 5, x ∈ invariant)
+
+    #transition "approach" -> "attempt"
+    add_transition!(automaton, 1, 2, 1)
+    guard = HalfSpace(sparsevec([x], [-1.], n), 100.) # x >= -100
+    t1 = @map(x -> x, dim: n, x ∈ guard)
+
+
+    #transition "approach" -> "abort"
+    add_transition!(automaton, 1, 3, 2)
+    guard_time = HPolyhedron([HalfSpace(sparsevec([t], [-1.], n), -t_abort_lower),  # t >= t_abort_lower
+                              HalfSpace(sparsevec([t], [1.], n), t_abort_upper)])   # t <= t_abort_upper
+    t2 = @map(x -> x, dim: n, x ∈ guard_time)
+
+    #transition "attempt" -> "abort"
+    add_transition!(automaton, 2, 3, 3)
+    t3 = @map(x -> x, dim: n, x ∈ guard_time)
+
+    H = HybridSystem(automaton=automaton,
+                     modes=[approaching, attempt, aborting],
+                     resetmaps=[t1, t2, t3])
+
+    return InitialValueProblem(H, init)
+end
+
+#variables
+const x, y, vx, vy, t = 1:5
+const numvars = 5
+const tan30 = tand(30)
+
+LineOfSightCone = HPolyhedron([HalfSpace(sparsevec([x], [-1.], numvars), 100.),   # x >= -100
+                    HalfSpace(sparsevec([x, y], [tan30, -1.], numvars), 0.),  # y >= x tan(30°)
+                    HalfSpace(sparsevec([x, y], [tan30, 1.], numvars), 0.),   # -y >= x tan(30°)
+                   ])
+
+Target = Hyperrectangle(zeros(2), [0.2, 0.2]);
+
+# ### Auxiliary functions
+
+function line_of_sight(sol)
+    all_idx = findall(x -> x == 2, location.(sol))  # attempt
+    for idx in all_idx
+        verif = all(set(R) ⊆ LineOfSightCone for R in sol[idx][2:end])
+        !verif && return false
+    end
+    return true
+end
+
+function absolute_velocity(R::AbstractLazyReachSet)
+    vx = 3
+    vy = 4
+    vx2 = set(overapproximate(project(R, vars=(vx,)), Interval)).dat
+    vy2 = set(overapproximate(project(R, vars=(vy,)), Interval)).dat
+    sqrt(vx2 + vy2)
+end
+
+function absolute_velocity(R::TaylorModelReachSet)
+    Z = overapprximate(R, Zonotope)
+    absolute_velocity(Z)
+end
+
+function velocity_constraint(sol)
+    all_idx = findall(x -> x == 2, location.(sol)) # attempt
+    for idx in all_idx
+        #maximum velocity measured in m/min
+        verif = all(absolute_velocity(R) < 0.055 * 60. for R in sol[idx])
+        !verif && return false
+    end
+    return true
+end
+
+function target_avoidance(sol)
+    all_idx = findall(x -> x == 3, location.(sol)) # aborting
+    for idx in all_idx
+        verif = all(is_intersection_empty(set(Projection(R, [x, y])), Target) for R in sol[idx])
+        !verif && return false
+    end
+    return true
+end
+
+# ## Results
+
+# We use `TMJets` algorithm with five-order expansion in time and first order expansion
```

Comment:
We used nQ = 1 and nT = 5. The transition to the aborting
mode is handled by clustering the flowpipe into 29 boxes.